crates/web5/src/dids/methods/did_jwk.rs

```diff
@@ -1,117 +1,246 @@
+
         let jwk_string = serde_json::to_string(&public_jwk)?;
         let method_specific_id = general_purpose::URL_SAFE_NO_PAD.encode(jwk_string);
 
-        let uri = format!("did:jwk:{}", method_specific_id);
+        let did_uri = format!("did:jwk:{}", method_specific_id);
 
-        let did = Did::parse(&uri)?;
+        let did = Did::parse(&did_uri)?;
 
-        let verification_method_id = format!("{}#0", uri);
-
-        let verification_method = VerificationMethod {
-            id: verification_method_id.clone(),
-            r#type: "JsonWebKey".to_string(),
-            controller: uri.clone(),
-            public_key_jwk: public_jwk.clone(),
-        };
+        let verification_method_id = format!("{}#0", did_uri);
 
         let document = Document {
-            id: uri.clone(),
-            verification_method: vec![verification_method.clone()],
+            context: Some(vec!["https://www.w3.org/ns/did/v1".to_string()]),
+            id: did_uri.clone(),
+            verification_method: vec![VerificationMethod {
+                id: verification_method_id.clone(),
+                r#type: "JsonWebKey".to_string(),
+                controller: did_uri.clone(),
+                public_key_jwk: public_jwk.clone(),
+            }],
             authentication: Some(vec![verification_method_id.clone()]),
             assertion_method: Some(vec![verification_method_id.clone()]),
             capability_invocation: Some(vec![verification_method_id.clone()]),
             capability_delegation: Some(vec![verification_method_id.clone()]),
             ..Default::default()
         };
 
-        Ok(Self { did, document })
+        Ok(BearerDid {
+            did,
+            document,
+            key_manager,
+        })
     }
 
-    pub fn from_uri(uri: &str) -> Result<Self> {
-        let resolution_result = DidJwk::resolve(uri);
-
-        match resolution_result.document {
-            None => Err(match resolution_result.resolution_metadata.error {
-                None => MethodError::ResolutionError(ResolutionMetadataError::InternalError),
-                Some(e) => MethodError::ResolutionError(e),
-            }),
-            Some(document) => {
-                let did = Did::parse(uri)?;
-                Ok(Self { did, document })
-            }
+    pub fn resolve(uri: &str) -> ResolutionResult {
+        let did = match Did::parse(uri) {
+            Ok(d) => d,
+            Err(_) => return ResolutionResult::from_error(ResolutionMetadataError::InvalidDid),
+        };
+
+        let decoded_jwk = match general_purpose::URL_SAFE_NO_PAD.decode(did.id) {
+            Ok(dj) => dj,
+            Err(_) => return ResolutionResult::from_error(ResolutionMetadataError::InvalidDid),
+        };
+
+        let public_jwk = match serde_json::from_slice::<Jwk>(&decoded_jwk) {
+            Ok(pj) => pj,
+            Err(_) => return ResolutionResult::from_error(ResolutionMetadataError::InvalidDid),
+        };
```

Comment:
`.map_err()?` might be more concise than match statements here

Reply:
The return type is `ResolutionResult`, not `Result`. Might be a better option than I have right now tho